src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1085,6 +1132,23 @@ theorem ofInt_mul {n} (x y : Int) : BitVec.ofInt n (x * y) =
   apply eq_of_toInt_eq
   simp
 
+@[simp]
+theorem BitVec.mul_one {x : BitVec w} : x * (1#w) = x := by
+  apply eq_of_toNat_eq
+  simp [toNat_mul, Nat.mod_eq_of_lt x.isLt]
+
+@[simp]
+theorem BitVec.mul_zero {x : BitVec w} : x * (0#w) = (0#w) := by
+  apply eq_of_toNat_eq
+  simp [toNat_mul]
+
+theorem BitVec.mul_add {x y z : BitVec w} :
```

Comment:
Can you double-check the theorem name? Maybe it would require a dist_ somewhere?

Reply:
It's named after `Nat.mul_add`: 

```lean
protected theorem mul_add (n m k : Nat) : n * (m + k) = n * m + n * k :=
  Nat.left_distrib n m k
```

Reply:
Actually, would we then need a left_distrib and have the various instantiations?

Reply:
@tobiasgrosser it's unclear to me, I'm trying to develop minimal theory to get the recurrences through, as it should establish a baseline to then improve the status quo. I guess we might want to mirror the `Nat` development --- I'm not sure.

Reply:
I guess we can leave this as is for now.